erpnext/public/js/utils.js

```diff
@@ -48,31 +48,41 @@ $.extend(erpnext, {
 		return cint(frappe.boot.sysdefaults.allow_stale);
 	},
 
-	setup_serial_no: function() {
-		var grid_row = cur_frm.open_grid_row();
-		if(!grid_row || !grid_row.grid_form.fields_dict.serial_no ||
+	setup_serial_or_batch_no: function() {
+		let grid_row = cur_frm.open_grid_row();
+		if (!grid_row || !grid_row.grid_form.fields_dict.serial_no ||
 			grid_row.grid_form.fields_dict.serial_no.get_status()!=="Write") return;
 
-		var $btn = $('<button class="btn btn-sm btn-default">'+__("Add Serial No")+'</button>')
-			.appendTo($("<div>")
-				.css({"margin-bottom": "10px", "margin-top": "10px"})
-				.appendTo(grid_row.grid_form.fields_dict.serial_no.$wrapper));
-
-		var me = this;
-		$btn.on("click", function() {
-			let callback = '';
-			let on_close = '';
-
-			frappe.model.get_value('Item', {'name':grid_row.doc.item_code}, 'has_serial_no',
-				(data) => {
-					if(data) {
-						grid_row.doc.has_serial_no = data.has_serial_no;
-						me.show_serial_batch_selector(grid_row.frm, grid_row.doc,
-							callback, on_close, true);
-					}
+		let me = this;
+		let attachSelectorButton = (innerText, appendLoction) => {
+			let $btnDiv = $("<div>").css({"margin-bottom": "10px", "margin-top": "10px"})
+				.appendTo(appendLoction);
+			let $btn = $(`<button class="btn btn-sm btn-default">${innerText}</button>`)
+				.appendTo($btnDiv);
+
+			$btn.on("click", function() {
+				let callback = '';
+				let on_close = '';
+				me.show_serial_batch_selector(grid_row.frm, grid_row.doc,
+					callback, on_close, true);
```

Comment:
just put single quotes instead of callback and on_close and remove the row 64 and 65